x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/UnaryStringFunction.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.scalar.string;
+
+import org.elasticsearch.xpack.sql.expression.Expression;
+import org.elasticsearch.xpack.sql.expression.FieldAttribute;
+import org.elasticsearch.xpack.sql.expression.function.scalar.UnaryScalarFunction;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.definition.ProcessorDefinition;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.definition.ProcessorDefinitions;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.definition.UnaryProcessorDefinition;
+import org.elasticsearch.xpack.sql.expression.function.scalar.script.ScriptTemplate;
+import org.elasticsearch.xpack.sql.expression.function.scalar.string.StringProcessor.StringOperation;
+import org.elasticsearch.xpack.sql.tree.Location;
+import org.elasticsearch.xpack.sql.util.StringUtils;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static org.elasticsearch.xpack.sql.expression.function.scalar.script.ParamsBuilder.paramsBuilder;
+
+public abstract class UnaryStringFunction extends UnaryScalarFunction {
+
+    protected UnaryStringFunction(Location location, Expression field) {
+        super(location, field);
+    }
+
+    @Override
+    public boolean foldable() {
+        return field().foldable();
+    }
+
+    @Override
+    public Object fold() {
+        return operation().apply(field().fold());
+    }
+
+    @Override
+    protected TypeResolution resolveType() {
+        if (!childrenResolved()) {
+            return new TypeResolution("Unresolved children");
+        }
+
+        return field().dataType().isString() ? TypeResolution.TYPE_RESOLVED : new TypeResolution(
+                "'%s' requires a string type, received %s", operation(), field().dataType().esType);
+    }
+
+    @Override
+    protected final ProcessorDefinition makeProcessorDefinition() {
+        return new UnaryProcessorDefinition(location(), this, ProcessorDefinitions.toProcessorDefinition(field()),
+                new StringProcessor(operation()));
+    }
+
+    protected abstract StringOperation operation();
+    
+    @Override
+    protected ScriptTemplate asScriptFrom(FieldAttribute field) {
+        //TODO change this to use _source instead of the exact form (aka field.keyword for text fields)
```

Comment:
Is there an issue raised for this one?

Reply:
I'm not sure we should do that. I mean, we should file an issue, but I think we should be very careful about using `_source` in script queries.